cx2t/src/claw/wani/transformation/sca/Parallelize.java

```diff
@@ -472,6 +476,252 @@ private void transformForGPU(XcodeProgram xcodeml)
     Directive.generateRoutineDirectives(xcodeml, _fctDef);
   }
 
+  /**
+   * Apply partial CPU transformation.
+   * When we can fusion the loop and keep the scalar "as is", we will do that
+   * in order to reduce cache misses.
+   * In other case we will apply a classic CPU transformation {@link
+   * #transformForCPU(XcodeProgram)} which will wrap each statement in DO loop.
+   *
+   * @param xcodeml Current XcodeML program unit.
+   * @throws IllegalTransformationException If promotion of arrays fails.
+   */
+  private void transformPartialForCPU(XcodeProgram xcodeml)
+          throws IllegalTransformationException {
+    // Apply transformation only when there is a body to modify
+    if (Body.isEmpty(_fctDef.body())) {
+      return;
+    }
+    // Variables on each "depth" of the method
+    List<Set<String>> depthVars = new ArrayList<>();
+    depthVars.add(new HashSet<String>());
+    // A growing list of affecting variables. Starting from the ones
+    // declared by SCA and successively any variables affected by the SCA
+    // declaration
+    Set<String> affectingVars = new HashSet<>(_promotions.keySet());
+    affectingVars.addAll(_arrayFieldsInOut);
+    // Keep track of the block we are in
+    int depth = 0;
+
+    // Extract all the variables used, this doesn't include vector's
+    // iterator variables
+    transformPartialForCPUAnalysis(_fctDef.body(), depth, depthVars,
+            affectingVars);
+
+    // Find the block we need to transform
+    Set<String>[] targetDepthIntersections = new Set[depthVars.size()];
```

Comment:
Array of generic are a bad idea, or 'unsafe' as the compiler put it.
Should be replaced by a List.

Reply:
If so, change it directly.